### PR TITLE
Thor

### PR DESCRIPTION
**Pull request by jermartin77**
I wish this pulled in your commit commments. So I updated the components to use a picture component and even a basic component that loads the content for the callouts and the banners. I'll add some notes inline because I don't know where some of this should go. We should also probably settle on our CSS structure before we go too deep.

### CODE REVIEW

File: assets/styles/core/_layout.scss [hunk 1]
```diff
@@ -16,7 +16,7 @@ html, body {
 }
 
 main {
-  padding-top: $header-height;
+  // padding-top: $header-height;
```
Comment (jermartin77) [inline assets/styles/core/_layout.scss]: Im getting rid of the fixed header for now, so it is a little more generic.

File: assets/styles/core/_theme.scss [hunk 2]
```diff
@@ -0,0 +1,28 @@
+.theme-light {
```
Comment (jermartin77) [inline assets/styles/core/_theme.scss]: I think this theme file could also house some predefined theme colors. Like black backgrround blue background, light gray background.

File: components/banners/BannerContent.vue [hunk 3]
```diff
@@ -0,0 +1,44 @@
+<template>
```
Comment (jermartin77) [inline components/banners/BannerContent.vue]: This file should serve for the content layer for videos, banner images and callouts.

File: components/banners/HeroBanner.vue [hunk 4]
```diff
@@ -1,22 +1,28 @@
...
-        {{ subtitle }}
-      </h3>
-    </div>
-    <picture>
-      <source :srcset="desktopUrl" media="(min-width: 800px)">
-      <img :src="mobileUrl" :alt="imageAlt">
-    </picture>
+    <banner-content
+                :title="title"
+                :content="subtitle">
```
Comment (jermartin77) [inline components/banners/HeroBanner.vue]: The `content` prop right here should match. I almost thing we build some basic pure bricks pages and then some basic content pages. I'm not sure about that yet.

File: components/banners/HeroBanner.vue [hunk 5]
```diff
@@ -27,27 +33,46 @@
...
   }
+
+
+
+
 </script>
 
-<style scoped lang="scss">
+<style lang="scss">
   .hero-banner {
```
Comment (jermartin77) [inline components/banners/HeroBanner.vue:48]: it's not scoped here so the child `BannerContent` component can inherit styles. I don't know if that is right.

File: components/slices/ImageBanner.vue [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+       // title: "color-quote"
+      }
+    },
+    props: {
+      fields: Object
+    }
+  }
+</script>
+
+<style lang="scss">
```
Comment (jermartin77) [inline components/slices/ImageBanner.vue]: Not scoped so child content can inherit styles.

**Comment by jermartin77** [comment]
Other notes. In the `ImageBanner` in prismic I used repeatable fields for links (like if you needed 2 buttons) I didn't see those come in in the data. 

Can't get the text as HTML stuff working. 

I'd like to use the headings selected in prismic and have those render in HTML.

File: components/banners/BannerContent.vue (hunk 3 above)
Comment (cbaxter713) [reply components/banners/BannerContent.vue]: Nice, I like it

File: components/banners/HeroBanner.vue (hunk 5 above)
Comment (cbaxter713) [reply components/banners/HeroBanner.vue:48]: We don't want to do this.  There may be some cases where we don't want encapsulated css on a component, but this isn't one of them.  What needs to be done here is `banner-content` needs to be extended to take alignment props that you pass into it, which control it's own alignment css from within the component.  Wherever you use `banner-content` you can style the host element for things like positioning, z-index, etc...but if you need styles of the internals, those should be handled by passing down props.

Any given component should have ZERO knowledge of any other components (i.e. classes used within, etc).  Each component is an API, with props that are passed in and events that are emitted out...thats it

File: components/utility/PictureComponent.vue [hunk 7]
```diff
@@ -0,0 +1,25 @@
+<template>
+  <picture class="picture-component">
+    <source :srcset="desktopUrl" media="(min-width: 800px)">
```
Comment (cbaxter713) [inline components/utility/PictureComponent.vue:3]: There should also be a `desktopBreakpoint` prop, which could have a default value of 800, but could be configurable to any value wherever this component is used

**Review by cbaxter713 — COMMENTED** [review]
I made a commit to this branch to get things working.  Take a look, just some minor cleanup of prop types and usage of prismic-dom: https://github.com/cbaxter713/nuxt-prismic-template/pull/4/commits/a3a7d96d9a8308afe366a95bf64ca8c586c99ea2 .  

Looking pretty good man!  Let's chat about this in the morning.  I do disagree about offloading the decision of h1 vs h2 vs h3 for titles to the content creator...I think that is a bad idea.  Too much room for error for there to be multiple h1's on the page, and could veer too far from the design...but lets chat about it.

Also, I think we need to figure out the best way to have no application logic in any display components, as in any usage of prismic-dom or anything like that.  Ideally all of our slices, banner components, etc will have their props setup to take in exactly what they need for display, and the Pages that compose those components will be responsible for using prismic-dom or anything else to configure the data into the form it needs to be for those display components.  We really need to do that to get into a habit of making truly reusable components, and isolate all application logic to the state management and top level Page components only

File: components/slices/ImageBanner.vue (hunk 6 above)
Comment (cbaxter713) [reply components/slices/ImageBanner.vue]: I think this should be scoped, and then don't style dom elements that live in the child components...but add classes to your child components and style those directly

File: components/utility/PictureComponent.vue [hunk 8]
```diff
@@ -0,0 +1,25 @@
+<template>
```
Comment (cbaxter713) [inline components/utility/PictureComponent.vue:1]: Don't need this component anymore, lets use the BasePicutre

File: components/banners/HeroBanner.vue [hunk 9]
```diff
@@ -1,53 +1,75 @@
...
-    <picture>
-      <source :srcset="desktopUrl" media="(min-width: 800px)">
-      <img :src="mobileUrl" :alt="imageAlt">
-    </picture>
+    <banner-content
+                :title="$prismic.asHtml(title)"
+                :content="$prismic.asHtml(content)"
+                class="theme-dark">
+    </banner-content>
+    <picture-component
```
Comment (cbaxter713) [inline components/banners/HeroBanner.vue]: use the BasePicture component here

**Pull request merged**